src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets

```
@@ -195,12 +195,38 @@
 
   <Target Name="_CheckContainersPackage" AfterTargets="Build">
     <PropertyGroup>
-      <ContainersPackageIdentity>Microsoft.NET.Build.Containers</ContainersPackageIdentity>
+      <_ContainersPackageIdentity>Microsoft.NET.Build.Containers</_ContainersPackageIdentity>
+      <_WebDefaultSdkVersion>7.0.300</_WebDefaultSdkVersion>
+      <_WorkerDefaultSdkVersion>8.0.100</_WorkerDefaultSdkVersion>
+      <_ConsoleDefaultSdkVersion>8.0.200</_ConsoleDefaultSdkVersion>
+      <_SdkCanPublishWeb>$([MSBuild]::VersionGreaterThanOrEquals('$(NETCoreSdkVersion)', '$(_WebDefaultSdkVersion)'))</_SdkCanPublishWeb>
+      <_SdkCanPublishWorker>$([MSBuild]::VersionGreaterThanOrEquals('$(NETCoreSdkVersion)', '$(_WorkerDefaultSdkVersion)'))</_SdkCanPublishWorker>
+      <_SdkCanPublishConsole>$([MSBuild]::VersionGreaterThanOrEquals('$(NETCoreSdkVersion)', '$(_ConsoleDefaultSdkVersion)'))</_SdkCanPublishConsole>
     </PropertyGroup>
     <ItemGroup>
-      <ContainersPackage Include="@(PackageReference)" Condition="'%(Identity)' == '$(ContainersPackageIdentity)'" />
+      <_ContainersPackage Include="@(PackageReference)" Condition="'%(Identity)' == '$(_ContainersPackageIdentity)'"/>
+      <_WebCapability Include="@(ProjectCapability)" Condition="'%(Identity)' == 'DotNetCoreWeb'" />
+      <_WorkerCapability Include="@(ProjectCapability)" Condition="'%(Identity)' == 'DotNetCoreWorker'" />
     </ItemGroup>
-    <Warning Text="Microsoft.NET.Build.Containers NuGet package is explicitly referenced. Consider removing the package reference to Microsoft.NET.Build.Containers as it is now part of .NET SDK." Condition="'@(ContainersPackage)' != ''" />
+
+    <PropertyGroup>
+      <_ContainerPackageIsPresent>false</_ContainerPackageIsPresent>
+      <_ContainerPackageIsPresent Condition="'@(_ContainersPackage)' != ''">true</_ContainerPackageIsPresent>
```

Can this simplify to

```suggestion
      <_ContainerPackageIsPresent Condition="'@(PackageReference->AnyHaveMetadataValue('Identity', '$(_ContainersPackageIdentity)')) != ''">true</_ContainerPackageIsPresent>
```

(and then maybe inline it?)